### PR TITLE
Introduce `EmbeddedDocument`

### PR DESCRIPTION
**Pull request by OskarStark**
Can be merged after
* #85

### CODE REVIEW

File: src/Document/EmbeddedDocument.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            $this->text,
+            $vector,
+            $this->metadata,
+        );
+    }
+
+    public function hasVector(): bool
+    {
+        return null !== $this->vector;
+    }
```
Comment (OskarStark) [inline src/Document/EmbeddedDocument.php:33]:
```suggestion

```

**Comment by chr-hertel** [comment]
what do you think about like a tupel of document classes:

`TextDocument` and `VectorDocument` - both have an UUID and Metadata

```php
/**
 * @param TextDocument|TextDocument[] $documents
 */
DocumentEmbedder::create(TextDocument|array $documents, ...): void

// only converting than

$vectorDocuments[] = new VectorDocument($vectors[$i], $document->id, $document->metadata);

// and drop text in store context completely

/**
 * @param VectorDocument|VectorDocument[] $documents
 **/
StoreInterface::add(VectorDocument|array $documents): void;

/**
 * @return VectorDocument[]
 */
VectorStoreInterface::query(Vector $vector, array $options = []): array;
```

**Comment by OskarStark** [comment]
I like it, I thought sth like this without inheritance too in the beginning

**Comment by OskarStark** [comment]
No I don't think we need that abstraction upstream

**Comment by chr-hertel** [comment]
you want to tackle that with your PRs or shall I?

**Comment by OskarStark** [comment]
If you have time, go ahead. Either use my PR or open a new one where you do the transformation completely

**Comment by OskarStark** [comment]
Superseded by
* #110

**Pull request closed without merging**